src/main/java/com/amihaiemil/versioneye/MkVersionEye.java

```diff
@@ -88,7 +106,29 @@ public VersionEye trusted() throws IOException {
 
     @Override
     public Me me() {
+        if(this.authenticated == null) {
+            throw new IllegalStateException("No Authenticated User.");
+        }
         return new MkMe(this.server, this.authenticated);
```

Comment:
@SherifWaly Can you add a puzzle for refactoring of ``MkMe``? Since it takes ``this.server`` (which contains all the authenticated users), it shouldn't take ``this.authenticated`` anymore, but only the username. And when it needs to return the authenticated json, it will search in the server by the username :) 

Then this class this class will also get rid of ``this.authenticated`` and retain only the username